Automattic/wp-calypso
# Refactor checkout sidebar upsell for Jetpack 2 year plans.

This PR updates the checkout page to accommodate the Jetpack 2-year plans (Backup T1 Bi-Yearly & Security T1 Bi-Yearly).


Asana Task: 1201210512993648-as-1204659654942251/f

## Proposed Changes

* Makes the intro pricing text show in the variant dropdown item for 2 year plans.
* Makes the checkout sidebar billing-term upsell box display the correct discount percentage and better displays the comparison to make more sense.
* Some small CSS updates to accommodate these changes.

## Testing Instructions

- First, sandbox `public-api.wordpress.com`.
- In your sandbox, open file: `wp-content/admin-plugins/wpcom-billing/default-shopping-cart-json-serializer.php`.
- Comment out lines 1050 - 1052 and save file, so that we can view Jetpack **Bi-Yearly** products at checkout.  It should looks like this:
```
// if ( 'jetpack' === $product->product_type && $product->renews_biyearly() ) {
// 	return null;
// }
```
- Now, checkout this PR/branch and spin up Calypso: `git fetch && git checkout update/checkout-sidebar-jetpack-2-year-plans && yarn start`
- Go to https://wordpress.com/checkout/jetpack/jetpack_backup_t1_yearly
- Open the billing-term variant dropdown and verify you see the "2 year" option and the intro pricing text says, "$143.05 for first 2 years then $238.80 per 2 years". (See screenshot)
- With the "One year" option selected, verify you see the sidebar billing-term upsell and that it makes sense. (See screenshot)
- Click the upsell CTA ("Switch to a two year plan"), and verify the 2 year variant gets selected.
- Perform the same above checks for https://wordpress.com/checkout/jetpack/jetpack_security_t1_yearly

### Screenshots

**BEFORE** | **AFTER**
--- | ---
![Markup 2023-05-25 at 09 35 34](https://github.com/Automattic/wp-calypso/assets/11078128/0b160daf-b75a-4b4a-b2c2-61b70844d21a) | ![Markup 2023-05-25 at 09 33 41](https://github.com/Automattic/wp-calypso/assets/11078128/e3e29b8c-d177-4d3b-85d9-042b2b9817e6)

--------

**BEFORE** | **AFTER**
--- | ---
![Markup 2023-05-25 at 09 39 55](https://github.com/Automattic/wp-calypso/assets/11078128/2bf4c84b-dbb2-456f-a5e3-8ce8f491cfa0) | ![Markup 2023-05-25 at 09 41 47](https://github.com/Automattic/wp-calypso/assets/11078128/4facaf02-506a-46c4-80db-2065df2faf71)

---------

**WPCOM Upselll comparison**

![Markup 2023-05-25 at 09 46 00](https://github.com/Automattic/wp-calypso/assets/11078128/c2e60fce-2f74-448e-92d8-c88a9acbf79b)






## Pre-merge Checklist
